src/app/store/effects/search.effects.ts

```diff
@@ -42,7 +42,7 @@ export class SearchEffects {
         item => item.id === SearchOptionIds.Libraries
       );
       const librarySelected = !!libItem && libItem.value;
-
+      action.payload = encodeURIComponent(action.payload);
       if (librarySelected) {
         this.router.navigateByUrl('/search-libraries;q=' + action.payload);
```

Comment:
I suggest converting the value on the fly so that payload value remains original (in case some other effect is listening on the same action).
```suggestion
        this.router.navigateByUrl('/search-libraries;q=' + encodeURIComponent (action.payload));
```

Reply:
You also missed the "else" clause that also runs "router.navigate"

Reply:
I have updated it!

Reply:
Excellent, thanks!